File: storage/db-core/src/main/java/com/coffee/gu/coupon/CouponEntity.java
package com.coffee.gu.coupon;

import com.coffee.gu.enums.CouponType;
import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Table(name = "coupon")
@Entity
public class CouponEntity extends BaseEntity {

    private String name;
    private CouponType type;
    private BigDecimal discount;
    private LocalDateTime expiredAt;

    public CouponEntity() {}

    public CouponEntity(String name, CouponType type, BigDecimal discount, LocalDateTime expiredAt) {
        this.name = name;
        this.type = type;
        this.discount = discount;
        this.expiredAt = expiredAt;
    }

    public Coupon toModel() {
        return new Coupon(id, name, type, discount, expiredAt);
    }

    public String getName() {
        return name;
    }

    public CouponType getType() {
        return type;
    }

    public BigDecimal getDiscount() {
        return discount;
    }

    public LocalDateTime getExpiredAt() {
        return expiredAt;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/coupon/CouponEntity.kt
package com.coffee.gu.coupon

import com.coffee.gu.BaseEntity
import com.coffee.gu.enums.CouponType
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal
import java.time.LocalDateTime

@Table(name = "coupon")
@Entity
class CouponEntity @JvmOverloads constructor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val name: String,
    @Enumerated(EnumType.STRING)
    val type: CouponType,
    val discount: BigDecimal,
    val expiredAt: LocalDateTime,
) : BaseEntity() {

    fun toModel(): Coupon {
        return Coupon(
            id = id,
            name = name,
            type = type,
            discount = discount,
            expiredAt = expiredAt,
        )
    }
}
